.travis.yml

```diff
@@ -65,78 +64,64 @@ matrix:
     fast_finish: true
 
     include:
-        - env: PYTHON_VERSION=2.7 SETUP_CMD='egg_info'
-        - env: PYTHON_VERSION=3.4 SETUP_CMD='egg_info'
         - env: PYTHON_VERSION=3.5 SETUP_CMD='egg_info'
         - env: PYTHON_VERSION=3.6 SETUP_CMD='egg_info'
+        - env: PYTHON_VERSION=3.7 SETUP_CMD='egg_info'
 
-        # Do a PEP8/pyflakes test with flake8
+        # PEP8/pyflakes test with flake8
         - env: MAIN_CMD="flake8 photutils --count $FLAKE8_OPT" SETUP_CMD=''
 
-        - env: SETUP_CMD='test --coverage'
-
-        # Check for sphinx doc build warnings - we do this first because it
-        # may run for a long time
-        - stage: Test docs, astropy dev, and without optional dependencies
-          env: PYTHON_VERSION=2.7 SETUP_CMD='build_docs -w'
-               CONDA_DEPENDENCIES='Cython scipy scikit-image scikit-learn matplotlib jupyter'
+        # All tests (with remote data) and coverage checks with the
+        # latest Python version.  The duration of the slowest 50 tests
+        # will also be listed.
+        - env: SETUP_CMD='test --coverage --remote-data=astropy -a "--durations=50"'
 
-        - stage: Test docs, astropy dev, and without optional dependencies
+        # Check for Sphinx doc build warnings
+        - stage: Comprehensive tests
           env: SETUP_CMD='build_docs -w'
-               CONDA_DEPENDENCIES='Cython scipy scikit-image scikit-learn matplotlib jupyter'
-               EVENT_TYPE='pull_request push cron'
+               EVENT_TYPE='push pull_request cron'
+
+        # Test with optional dependencies disabled
+        - stage: Comprehensive tests
+          env: CONDA_DEPENDENCIES=$CONDA_REQUIRED_DEPENDENCIES
 
-        # Now try Astropy dev and LTS versions with the latest Python.
-        # We disable LTS testing until astropy 3.0 is out, as the current
-        # stable is also the LTS
-        - stage: Test docs, astropy dev, and without optional dependencies
+        # Test with Astropy dev version and the latest Python
+        - stage: Comprehensive tests
           env: ASTROPY_VERSION=development
                EVENT_TYPE='pull_request push cron'
 
-        #- stage: Test docs, astropy dev, and without optional dependencies
-        #  os: linux
-        #  env: PYTHON_VERSION=3.5 ASTROPY_VERSION=lts
+        # Test with Astropy LTS version and the latest Python
+        # NOTE:  currently disabled because Astropy stable is also the LTS
```

Comment:
Add back the LTS testing here since 3.0.x is out

Reply:
But one can use LTS with 3.5+, too, even though it's difficult to see any scenario when one has to stuck with LTS but can use the latest and greatest photutils.
Anyway, I wasn't simply sure whether it's an oversight or you're thinking of requiring astropy 3.0+ for photutils.

Reply:
Good point.  I hadn't thought about the minimum astropy version before.  I think I'm inclined to require 3.0+.  Does that sound reasonable?

Reply:
Yes, I think that's reasonable, but needs to be stated.

Reply:
Yes, I'll add it in the installation docs (requirements).

Reply:
Many thanks for that info @moustakas!  In that case I'll leave the dependency as astropy 2+.  Astropy 3+ isn't required *yet* (i.e. photutils isn't using any Astropy 3+ specific features yet), but I imagine that will change.

Are you using Python >= 3.5?  That will be the minimum requirement for photutils 0.5.

As a side note, you also may want to consider upgrading to astropy 3.1 (scheduled for Nov. 2018) for the many performance improvements.

Reply:
@larrybradley - It this case, please add a build for LTS.